shell/edit/workload/mixins/workload.js

```diff
@@ -184,25 +184,17 @@ export default {
           name:            `container-0`,
         }];
 
-        const metadata = { ...this.value.metadata };
-
-        const podSpec = { template: { spec: { containers: podContainers, initContainers: [] }, metadata } };
+        const podSpec = {
+          containers:     podContainers,
+          initContainers: [],
+        };
 
         this.$set(this.value, 'spec', podSpec);
       }
     }
 
-    // EDIT view for POD
-    // Transform it from POD world to workload
-    if ((this.mode === _EDIT || this.mode === _VIEW || this.realMode === _CLONE ) && this.value.type === 'pod') {
-      const podSpec = { ...this.value.spec };
-      const metadata = { ...this.value.metadata };
-
-      this.$set(this.value.spec, 'template', { spec: podSpec, metadata });
-    }
-
     const spec = this.value.spec;
-    let podTemplateSpec = type === WORKLOAD_TYPES.CRON_JOB ? spec.jobTemplate.spec.template.spec : spec?.template?.spec;
+    let podTemplateSpec = type === WORKLOAD_TYPES.CRON_JOB ? spec.jobTemplate.spec.template.spec : this.value.type === POD ? spec : spec?.template?.spec;
```

Comment:
I might consider rewriting this so that it's much more clear that we have multiple conditionals to deal Pods, Cron job, and the rest.